src/ExtendedTable.svelte

```diff
@@ -97,39 +98,66 @@
         clearCaches();
     };
 
-    const getClasses = (classes, index, prefix) => {
+    const getOddEvenClass = (index, prefix) => {
         if (index % 2 === 0) {
-            classes.push('even');
-        } else {
-            classes.push('odd');
+            return `${prefix}-even`;
         }
 
-        return classes.map((c) => prefix + '-' + c)
-            .join(" ");
+        return `${prefix}-odd`;
     };
 
-    const slugEx = new RegExp(/[^a-z0-9]/g);
+    const slugEx = new RegExp(/[^a-z0-9\-]/ig);
     const sluggify = (input) => {
-        return input.replace(slugEx, '_');
+        if (!input) {
+            return
+        }
+        return ("" + input).replace(slugEx, '_');
```

Comment:
Replacing spaces with '_' prevents multiple classes to be returned by the same function. If we want to support that, the 2 solutions can be:
1. Don't replace spaces with '_'
2. Support arrays as return value and concat it if type is array.

Reply:
Both ways would indeed be possible. I'll try to outline pros and cons here:

### Not replacing spaces (supporting string-arrays)

On the pro side this feels pretty normal and is in a way unexpected, that we transform the result of the callback.

### Support real arrays

This also feels pretty normal, as we are here in a JavaScript function, so why shouldn't we return an array?

The downside for both is the additional complexity (in code and for the user) and that we (if we would support both ways) allow multiple return types.

### Future perspective

I tend to say that we wait for someone who needs this feature. Or do you need it today?
Looking into the future, if we defer this feature for now, it might count as a small bc break if the implement this later on (and omit replacing spaces with underscores). Introducing arrays seems not to infringe backwards compatibility.

Do you have a preference?

Reply:
Another way would be to remove sluggification for values returned from the callback.. IMHO, sluggifying (or is it sluggification?) makes most sense with the `propertyPath` (which might be out of control of the dev). Removing it might fit the principle of least surprise best I guess..

Reply:
Good thoughts. 

I think sluggifying only `propertyPath` makes sense since that is non-dev control like you said and it makes sense to sluggify it as there is very little to zero chance devs wants spaces as class separators. `value` function on the other hand, devs would normally expect spaces to be honored, so I sluggifying it would certainly surprise them.

So I would go with only sluggifying `propertyPath` and not `value` functions.

Reply:
Thanks for the feedback. I removed sluggifying the `value` and added some documentation.